slyd/app/utils/ferry-websocket.js

```diff
@@ -0,0 +1,215 @@
+import Ember from 'ember';
+import config from '../config/environment';
+import ApplicationUtils from '../mixins/application-utils';
+
+/* global URI */
+
+const DEFAULT_RECONNECT_TIMEOUT = 5000;
+const DEFAULT_MAX_RECONNECT_TIMEOUT = 60000;
+const DEFAULT_COMMANDS = Object.freeze({
+    heartbeat: function() {},
+    saveChanges: function() {},
+    delete: function() {},
+    rename: function() {},
+    resolve: function() {},
+});
+
+var defaultUrl = function() {
+    var uri = URI.parse(config.SLYD_URL || window.location.protocol + '//' + window.location.host);
+    if (!/wss?/.test(uri.protocol)) {
+        uri.protocol = uri.protocol === 'https' ? 'wss' : 'ws';
+    }
+    uri.path = '/ws';
+    return URI.build(uri);
+};
+
+export default Ember.Object.extend(ApplicationUtils, {
+    init: function(options) {
+        options = options || {};
+        this.set('commands', options.commands || {});
+        this.set('url', options.url || defaultUrl());
+        this.set('cleanup', options.cleanup || function() {});
+        this.set('init', options.init || function() {});
+        this.set('protocols', options.protocols);
+        this.set('commands', this._buildCommands(options.commands || {}));
+        this.set('ws', null);
+        this.set('reconnectTimeout', DEFAULT_RECONNECT_TIMEOUT);
+        this.set('closed', true);
+        this.set('opened', false);
```

Comment:
Use `Ember.computed.not` to avoid having to set both `closed` and `opened` all the time.